### ISSUE TITLE
Document that variable needs to be read for --allow-redefinition to work

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#6963]
See #6959.

Part of the motivation is that this shouldn't trigger redefinition:

```py
x = None  # type: List[int]
...  # Do nothing with `x`
x = []  # Want to preserve type context
```

**Comment by Shivansh-khare** [comment]
I didn't get the issue can you tell in brief.

**Comment by terencehonles** [comment]
There look like there are other constraints that should be added where this is documented. Some quick examples is in a try/catch block, if a function is defined in the same scope #10016, ...

**Comment by AlexWaygood** [comment]
> [Current documentation](https://mypy.readthedocs.io/en/latest/command_line.html#cmdoption-mypy-allow-redefinition) says:
> 
> > The variable must be used before it can be redefined:
> 
> Can we close this? @AlexWaygood

Yup, fixed by @KotlinIsland in #11951! 🥳
